repair/repair.cc

```diff
@@ -2052,6 +2052,77 @@ future<> repair_service::replace_with_repair(locator::token_metadata_ptr tmptr,
     co_return co_await do_rebuild_replace_with_repair(std::move(cloned_tmptr), std::move(op), myloc.dc, reason, std::move(ignore_nodes));
 }
 
+// Repair all tablets belong to this node for the given table
+future<> repair_service::repair_tablets(repair_uniq_id rid, sstring keyspace_name, sstring table_name, host2ip_t host2ip, bool primary_replica_only) {
+    auto& t = _db.local().find_column_family(keyspace_name, table_name);
+    if (!t.uses_tablets()) {
+        throw std::runtime_error(format("Table {}.{} is not a tablet table", keyspace_name, table_name));
+    }
+    table_id tid = t.schema()->id();
+    // FIXME: we need to wait for current tablet movement and disable future tablet movment
+    auto erm = t.get_effective_replication_map();
+    auto& tmap = erm->get_token_metadata_ptr()->tablets().get_tablet_map(tid);
```

Comment:
We should refresh the erm after each tablet to give the load balancer a chance to make progress.

I don't mind if this comes as a follow-up, but it's important.

Reply:
Noted. I will consider this as follow up. It is more complicated than refreshing. The tablet mapping (both number of tablets and ownership of tablet) might change after we have repaired some of the tablets. We need to figure out where to continue after the refresh.

Reply:
Suggest to maintain an interval map of unrepaired ranges. Start from the beginning, if a range is unowned by this node, remove it. If a range is owned by a shard and the shard isn't repairing, remove the range and start repairing. If the range is owned by a shard and the shard is repairing, move on to the next range.